router-sink/src/main/java/org/springframework/cloud/stream/module/router/sink/RouterSink.java

```diff
@@ -0,0 +1,95 @@
+import org.springframework.cloud.stream.binding.BinderAwareChannelResolver;
+import org.springframework.cloud.stream.config.SpelExpressionConverterConfiguration;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.module.common.ScriptVariableGeneratorConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.groovy.GroovyScriptExecutingMessageProcessor;
+import org.springframework.integration.router.AbstractMappingMessageRouter;
+import org.springframework.integration.router.ExpressionEvaluatingRouter;
+import org.springframework.integration.router.MethodInvokingRouter;
+import org.springframework.integration.scripting.RefreshableResourceScriptSource;
+import org.springframework.integration.scripting.ScriptVariableGenerator;
+import org.springframework.scripting.ScriptSource;
+
+/**
+ * A sink module that routes to one or more named channels.
+ *
+ * @author Gary Russell
+ */
+@EnableBinding(Sink.class)
+@Import({ SpelExpressionConverterConfiguration.class, ScriptVariableGeneratorConfiguration.class })
+@EnableConfigurationProperties(RouterSinkProperties.class)
+public class RouterSink {
+
+	@Autowired
+	private RouterSinkProperties properties;
+
+	@Autowired
+	private ScriptVariableGenerator scriptVariableGenerator;
+
+	@Autowired
+	@Bindings(RouterSink.class)
+	Sink channels;
+
+	@Bean
+	@ServiceActivator(inputChannel=Sink.INPUT)
+	public AbstractMappingMessageRouter router(BinderAwareChannelResolver channelResolver,
+			ConfigurableBeanFactory beanFactory) throws Exception {
+		AbstractMappingMessageRouter router;
+		if (this.properties.getScript() != null) {
+			router = new MethodInvokingRouter(scriptProcessor(beanFactory));
+		}
+		else {
+			router = new ExpressionEvaluatingRouter(this.properties.getExpression());
+		}
+		router.setDefaultOutputChannelName(this.properties.getDefaultOutputChannel());
+		router.setResolutionRequired(this.properties.isResolutionRequired());
+		Map<String, String> mappings = new LinkedHashMap<String, String>();
+		int n = 0;
+		for (String value : this.properties.getValues()) {
+			mappings.put(value, this.properties.getDestinations()[n++]);
+		}
+		router.setChannelMappings(mappings);
+		router.setChannelResolver(channelResolver);
+		return router;
+	}
+
+	private GroovyScriptExecutingMessageProcessor scriptProcessor(ConfigurableBeanFactory beanFactory) throws Exception {
+		ScriptSource scriptSource = new RefreshableResourceScriptSource(this.properties.getScript(),
+				this.properties.getRefreshDelay());
+		GroovyScriptExecutingMessageProcessor processor = new GroovyScriptExecutingMessageProcessor(scriptSource,
+				this.scriptVariableGenerator);
+		processor.setBeanFactory(beanFactory);
+		processor.setBeanClassLoader(beanFactory.getBeanClassLoader());
```

Comment:
Can't `GroovyScriptExecutingMessageProcessor` be represent as a `@Bean` as well?
Right, with some `@ConditionalOnProperty("script")` anyway do not create it for nothing.

The current approach is fine, too. Just in case to share some mind and gather more info.